org.scala-ide.sdt.core.tests/src/org/scalaide/core/testsetup/SDTTestUtils.scala

```diff
@@ -293,17 +301,33 @@ object SDTTestUtils extends HasLogger {
         override lazy val project = scalaProject
       }
       projectSetup.project.presentationCompiler { c => f(c) }
-    }
-    finally deleteProjects(projectSetup.project)
+    } finally deleteProjects(projectSetup.project)
   }
 
   /** Create a project in the current workspace. If `withSourceRoot` is true,
    *  it creates a source folder called `src`.
    */
   def createProjectInWorkspace(projectName: String, withSourceRoot: Boolean = true): IScalaProject =
-    internalCreateProjectInWorkspace(projectName, withSourceRoot)
+    internalCreateProjectInWorkspace(projectName, if (withSourceRoot) withSourceRootOnly else withNoSourceRoot)
+
+  def createProjectInWorkspace(projectName: String, withSrcOutputStructure: SrcPathOutputEntry): IScalaProject =
+    internalCreateProjectInWorkspace(projectName, withSrcOutputStructure)
+
+  type SrcPathOutputEntry = (IProject, IJavaProject, ArrayBuffer[IClasspathEntry]) => Unit
```

Comment:
I prefer to see `(IProject, IJavaProject, Seq[IClasspathEntry]) => Seq[IClasspathEntry]` and implementation that do not modify it's arguments.